python/ray/tests/kuberay/utils.py

```diff
@@ -60,14 +66,25 @@ def _get_num_pods(namespace: str) -> int:
 
     # If there aren't any pods, the output is any empty string.
     if not get_pod_output:
-        return 0
+        return []
     else:
-        return len(get_pod_output.split("\n"))
+        return get_pod_output.split("\n")
+    pass
 
 
-def wait_for_pod_to_start(pod: str, namespace: str, tries=60, backoff_s=5) -> None:
-    """Waits for the pod to enter running Running status.phase."""
+def wait_for_pod_to_start(
+    pod_name_filter: str, namespace: str, tries=60, backoff_s=5
+) -> None:
+    """Waits for a pod to have Running status.phase.
+
+    More precisely, waits until there is a pod with name containing `pod_name_filter`
+    and the pod has Running status.phase."""
     for i in range(tries):
+        try:
+            pod = get_pod(pod_name_filter=pod_name_filter, namespace=namespace)
+        except AssertionError:
```

Comment:
fyi: AssertionError seems like an odd choice of error for this -- it would probably make the most sense to return a list (and return an empty list if no pod is found), or if a single pod is usually suspected, either return that pod or None.

No need to change it for this PR, I'm just pointing it out :)

Reply:
Will change to return None when there's no pod.
(Will migrate the Optional from the return type of `kubectl_exec` to the return type of `get_pod`.)

Reply:
Fixed.